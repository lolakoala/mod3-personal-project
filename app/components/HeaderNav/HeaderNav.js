import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class HeaderNav extends Component {
  render() {
    const { currentUser, usersHouse, signOut, leaveHouse } = this.props;
    const userStuff = <div className='user-stuff'>
      <p className='greeting'>{`Welcome to ${usersHouse.houseName},  ${currentUser.name}.`}</p>
      <Link to="/houselist">Manage House</Link>
      <Link to="/userlist">My Lists</Link>
      <Link to='/' onClick={() => { leaveHouse(currentUser, usersHouse); }}>Leave House</Link>
      <Link to='/' onClick={signOut}>Sign Out</Link>
    </div>;
    return (
      <div>
        <Link to="/">
          <h1>Tribe</h1>
        </Link>
        <h3>A Communal Living Management App</h3>
        {usersHouse.houseName ? userStuff : null}
      </div>
    );
  }
}

export default HeaderNav;

HeaderNav.propTypes = {
  currentUser: PropTypes.object,
  usersHouse: PropTypes.object,
  signOut: PropTypes.func,
  leaveHouse: PropTypes.func
};
