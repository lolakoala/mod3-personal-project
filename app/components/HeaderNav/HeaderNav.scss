.app-header {
  width: auto;
  height: auto;
}

.logo {
  font-family: 'Shadows Into Light', cursive;
  display: flex;
  flex-direction: column;
  margin: 20px;
  background-color: rgba(141, 132, 73, 0.7);
  border-radius: 25px;
  height: 75px;
  width: 125px;
  height: 150px;
  padding: 10px;
  justify-content: center;
  align-items: center;
  color: #3d6aa2;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 100;
}

.logo h1 {
  width: 125px;
  text-align: center;
  color: #3d6aa2;
  line-height: normal;
  font-size: 50px;
  background-color: rgba(141, 132, 73, 0);
  border: none;
}

.logo h1:hover {
  cursor: pointer;
  color: #D83D59;
}

.logo h3{
  background-color: rgba(141, 132, 73, 0);
}

.app-header a {
  text-decoration: none;
}

.user-stuff-nav {
  height: auto;
  width: auto;
  margin: 15px;
  display: flex;
  flex-direction: column;
  position: fixed;
  top: 0;
  right: 0;
}

.user-stuff-nav .manage-house,
.user-stuff-nav .my-lists,
.user-stuff-nav .leave-house,
.user-stuff-nav .signout,
.user-stuff-nav .home {
  border: 2px solid #D83D59;  
  font-family: 'BenchNine', sans-serif;
  background-color: #D83D59;
  width: 120px;
  height: 25px;
  color: #F4CCE2;
  font-size: 20px;
  border-radius: 25px;
  text-align: center;
  margin: 5px;
}

.user-stuff-nav .manage-house:hover,
.user-stuff-nav .my-lists:hover,
.user-stuff-nav .leave-house:hover,
.user-stuff-nav .signout:hover,
.user-stuff-nav .home:hover,
.user-stuff-nav .active {
  cursor: pointer;
  border: 2px dashed #D83D59;
  background-color: #F4CCE2;
  color: #D83D59;
}

.greeting {
  display: inline-block;
  margin: 20px;
  height: auto;
  width: auto;
  font-size: 20px;
  color:  color: #3d6aa2;
  text-align: center;
  font-family: 'BenchNine', sans-serif;
  margin-top: 200px;
}
