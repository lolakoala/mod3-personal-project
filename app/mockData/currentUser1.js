const currentUser1 = {
  id : "RVde1m9BhkQBLZlTJNsFlQZXqP43",
  name: "Amanda Brenner",
  house: "Dog9"
};

export default currentUser1;
