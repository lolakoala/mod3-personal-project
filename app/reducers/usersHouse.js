import firebase from '../firebase.js';

const usersHouse = (state = {}, action) => {
  const housesRef = firebase.database().ref('houses');
  switch (action.type) {
  case 'CREATE_HOUSE':
    housesRef.push(action.house);
    return action.house;
  case 'GET_HOUSE':
    const updatedHouse = Object.assign(action.usersHouse, { users: [...action.usersHouse.users, action.user] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updatedHouse);
    return updatedHouse;
  case 'LOGIN_SUCCESS':
    return action.usersHouse;
  case 'SIGNOUT':
    return {};
  case 'LEAVE_HOUSE':
    const deleteUser = action.usersHouse.users.filter(user => user.id !== action.currentUser.id);
    const houseWithoutUser = Object.assign(action.usersHouse, { users: deleteUser });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithoutUser);
    return {};
  case 'ADD_BILL':
    const takeOutFakeBills = action.usersHouse.bills.filter(bill => bill.title !== 'fake');
    const houseWithBill = Object.assign(action.usersHouse, { bills: [...takeOutFakeBills, action.bill] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithBill);
    return houseWithBill;
  case 'ADD_BULLETIN':
    const takeOutFakeBulletins = action.usersHouse.bulletins.filter(bulletin => bulletin.title !== 'fake');
    const houseWithBulletin = Object.assign(action.usersHouse, { bulletins: [...takeOutFakeBulletins, action.bulletin] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithBulletin);
    return houseWithBulletin;
  case 'ADD_CHORE':
    const takeOutFakeChore = action.usersHouse.chores.filter(chore => chore.title !== 'fake');
    const houseWithChore = Object.assign(action.usersHouse, { chores: [...takeOutFakeChore, action.chore] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithChore);
    return houseWithChore;
  case 'ADD_READER':
    const oldBulletin = action.usersHouse.bulletins.find(bulletin => bulletin.id === action.bulletinId);
    if (oldBulletin.hasRead.includes(action.userId)) {
      return action.usersHouse;
    }
    const newBulletin = Object.assign(oldBulletin, { hasRead: [...oldBulletin.hasRead, action.userId] });
    const updateBulletins = action.usersHouse.bulletins.filter(bulletin => {
      return bulletin.id !== action.bulletinId;
    });
    const updateWithBulletin = Object.assign(action.usersHouse, { bulletins: [...updateBulletins, newBulletin] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updateWithBulletin);
    return updateWithBulletin;
  case 'MARK_PAID':
    const oldBill = action.usersHouse.bills.find(bill => bill.id === action.billId);
    const user = oldBill.allUsersTotals.find(user => user.id === action.userId);
    if (user.paid === true) {
      return action.usersHouse;
    }
    const newUser = Object.assign(user, { paid: true });
    const newAllUsersTotals = oldBill.allUsersTotals.filter(user => user.id !== action.userId);
    const newBill = Object.assign(oldBill, { allUsersTotals: [...newAllUsersTotals, newUser] });
    const updatedBills = action.usersHouse.bills.filter(bill => {
      return bill.id !== action.billId;
    });
    const updateWithBill = Object.assign(action.usersHouse, { bills: [...updatedBills, newBill] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updateWithBill);
    return updateWithBill;
  default:
    return state;
  }
};

export default usersHouse;
