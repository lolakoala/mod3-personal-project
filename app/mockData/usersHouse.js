import currentUser1 from './currentUser1.js';
import currentUser2 from './currentUser2.js';

const bill1 = {
  allUsersTotals: [
    {
      id: currentUser1.id,
      paid: false,
      total: 50
    },
    {
      id: currentUser2.id,
      paid: false,
      total: 50
    }
  ],
  datePosted: "11/01/2017",
  details: "We used a lot of water.",
  duedate: "11/9/17",
  id: "b3019762705779",
  parsedDuedate: 1510210800000,
  postedBy: currentUser1,
  title: "Water Bill",
  total: "100"
};

const bulletin1 = {
  datePosted: "11/01/2017",
  details: "The dogs got out through a hole in the fence yesterday.",
  hasRead: [currentUser1.id],
  id: 1509567098747,
  postedBy: currentUser1,
  title: "Don't leave dogs out unattended."
};

const chore1 = {
  assignedTo: currentUser1.id,
  datePosted: "10/31/2017",
  details: "Empty the dishwasher.",
  done: false,
  id: "c1509485241076",
  postedBy: currentUser1,
  title: "Dishes",
  urgency: "low"
};

const usersHouse = {
  houseKey: "-KxoIBWy5UYRscFYdWty",
  bills: [bill1],
  bulletins: [bulletin1],
  chores: [chore1],
  houseCode: "woof9",
  houseName: "Dog9",
  users: [currentUser1, currentUser2]
};

export default usersHouse;
