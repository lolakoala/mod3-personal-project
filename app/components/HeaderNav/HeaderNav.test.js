import React from "react";
import HeaderNav from './HeaderNav.js';
import { shallow } from "enzyme";
import usersHouse from '../../mockData/usersHouse.js';
import currentUser1 from '../../mockData/currentUser1.js';

describe('HeaderNav', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<HeaderNav
    currentUser={currentUser1}
    usersHouse={usersHouse}
    signOut={mockFn}
    leaveHouse={mockFn}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match snap', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a title and subtitle', () => {
    const title = wrapper.find('h1');
    const subtitle = wrapper.find('h3');

    expect(title.length).toEqual(1);
    expect(subtitle.length).toEqual(1);
  });

  it('should have five links', () => {
    const link = wrapper.find('Link');

    expect(link.length).toEqual(5);
  });

  it('should have signout and leavehouse div', () => {
    const signoutDiv = wrapper.find('.signout');
    const leavehouseDiv = wrapper.find('.leave-house');

    expect(signoutDiv.length).toEqual(1);
    expect(leavehouseDiv.length).toEqual(1);
  });

  it('should render userStuff', () => {
    const userStuff = wrapper.find('.user-stuff');
    const greeting = wrapper.find('.greeting');

    expect(userStuff.length).toEqual(1);
    expect(greeting.text()).toEqual(`Welcome to ${usersHouse.houseName},  ${currentUser1.name}.`);
  });

  it('should not render userStuff when no user', () => {
    const altWrapper = shallow(<HeaderNav
      currentUser={{}}
      usersHouse={{}}
      signOut={mockFn}
      leaveHouse={mockFn}/>);
    const userStuff = altWrapper.find('.user-stuff');

    expect(userStuff.length).toEqual(0);
  });

  it('should fire actions', () => {
    const signoutDiv = wrapper.find('.signout');
    const leavehouseDiv = wrapper.find('.leave-house');

    signoutDiv.simulate('click');
    leavehouseDiv.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(2);
  });
});
