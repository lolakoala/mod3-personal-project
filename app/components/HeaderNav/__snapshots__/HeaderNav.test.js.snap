// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderNav
    currentUser={
      Object {
        "house": "Dog9",
        "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
        "name": "Amanda Brenner",
      }
    }
    leaveHouse={[Function]}
    location={
      Object {
        "pathname": "/",
      }
    }
    signOut={[Function]}
    usersHouse={
      Object {
        "bills": Array [
          Object {
            "allUsersTotals": Array [
              Object {
                "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                "paid": false,
                "total": 50,
              },
              Object {
                "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                "paid": false,
                "total": 50,
              },
            ],
            "datePosted": "11/01/2017",
            "details": "We used a lot of water.",
            "duedate": "11/9/17",
            "id": "b3019762705779",
            "parsedDuedate": 1510210800000,
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Water Bill",
            "total": "100",
          },
        ],
        "bulletins": Array [
          Object {
            "datePosted": "11/01/2017",
            "details": "The dogs got out through a hole in the fence yesterday.",
            "hasRead": Array [
              "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            ],
            "id": 1509567098747,
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Don't leave dogs out unattended.",
          },
        ],
        "chores": Array [
          Object {
            "assignedTo": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "datePosted": "10/31/2017",
            "details": "Empty the dishwasher.",
            "done": false,
            "id": "c1509485241076",
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Dishes",
            "urgency": "low",
          },
        ],
        "houseCode": "woof9",
        "houseKey": "-KxoIBWy5UYRscFYdWty",
        "houseName": "Dog9",
        "users": Array [
          Object {
            "house": "Dog9",
            "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "name": "Amanda Brenner",
          },
          Object {
            "house": "Dog9",
            "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
            "name": "Lola Koala",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo"
        >
          <Link
            replace={false}
            to="/"
          >
            <h1>
              Tribe
            </h1>
          </Link>
          <h3>
            A Communal Living Management App
          </h3>
        </div>,
        <div
          className="user-stuff"
        >
          <div
            className="user-stuff-nav"
          >
            <Link
              className="home active"
              replace={false}
              to="/"
            >
              Home
            </Link>
            <Link
              className="manage-house "
              replace={false}
              to="/houselist"
            >
              Manage House
            </Link>
            <Link
              className="my-lists "
              replace={false}
              to="/userlist"
            >
              My Lists
            </Link>
            <Link
              className="leave-house"
              onClick={[Function]}
              replace={false}
              to="/"
            >
              Leave House
            </Link>
            <Link
              className="signout"
              onClick={[Function]}
              replace={false}
              to="/"
            >
              Sign Out
            </Link>
          </div>
          <p
            className="greeting"
          >
            Welcome to Dog9,  Amanda Brenner.
          </p>
        </div>,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
            >
              <h1>
                Tribe
              </h1>
            </Link>,
            <h3>
              A Communal Living Management App
            </h3>,
          ],
          "className": "logo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <h1>
                Tribe
              </h1>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Tribe",
              },
              "ref": null,
              "rendered": "Tribe",
              "type": "h1",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "A Communal Living Management App",
            },
            "ref": null,
            "rendered": "A Communal Living Management App",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="user-stuff-nav"
            >
              <Link
                className="home active"
                replace={false}
                to="/"
              >
                Home
              </Link>
              <Link
                className="manage-house "
                replace={false}
                to="/houselist"
              >
                Manage House
              </Link>
              <Link
                className="my-lists "
                replace={false}
                to="/userlist"
              >
                My Lists
              </Link>
              <Link
                className="leave-house"
                onClick={[Function]}
                replace={false}
                to="/"
              >
                Leave House
              </Link>
              <Link
                className="signout"
                onClick={[Function]}
                replace={false}
                to="/"
              >
                Sign Out
              </Link>
            </div>,
            <p
              className="greeting"
            >
              Welcome to Dog9,  Amanda Brenner.
            </p>,
          ],
          "className": "user-stuff",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="home active"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>,
                <Link
                  className="manage-house "
                  replace={false}
                  to="/houselist"
                >
                  Manage House
                </Link>,
                <Link
                  className="my-lists "
                  replace={false}
                  to="/userlist"
                >
                  My Lists
                </Link>,
                <Link
                  className="leave-house"
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  Leave House
                </Link>,
                <Link
                  className="signout"
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  Sign Out
                </Link>,
              ],
              "className": "user-stuff-nav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "className": "home active",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Manage House",
                  "className": "manage-house ",
                  "replace": false,
                  "to": "/houselist",
                },
                "ref": null,
                "rendered": "Manage House",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "My Lists",
                  "className": "my-lists ",
                  "replace": false,
                  "to": "/userlist",
                },
                "ref": null,
                "rendered": "My Lists",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Leave House",
                  "className": "leave-house",
                  "onClick": [Function],
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Leave House",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Out",
                  "className": "signout",
                  "onClick": [Function],
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Sign Out",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Dog9,  Amanda Brenner.",
              "className": "greeting",
            },
            "ref": null,
            "rendered": "Welcome to Dog9,  Amanda Brenner.",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo"
          >
            <Link
              replace={false}
              to="/"
            >
              <h1>
                Tribe
              </h1>
            </Link>
            <h3>
              A Communal Living Management App
            </h3>
          </div>,
          <div
            className="user-stuff"
          >
            <div
              className="user-stuff-nav"
            >
              <Link
                className="home active"
                replace={false}
                to="/"
              >
                Home
              </Link>
              <Link
                className="manage-house "
                replace={false}
                to="/houselist"
              >
                Manage House
              </Link>
              <Link
                className="my-lists "
                replace={false}
                to="/userlist"
              >
                My Lists
              </Link>
              <Link
                className="leave-house"
                onClick={[Function]}
                replace={false}
                to="/"
              >
                Leave House
              </Link>
              <Link
                className="signout"
                onClick={[Function]}
                replace={false}
                to="/"
              >
                Sign Out
              </Link>
            </div>
            <p
              className="greeting"
            >
              Welcome to Dog9,  Amanda Brenner.
            </p>
          </div>,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
              >
                <h1>
                  Tribe
                </h1>
              </Link>,
              <h3>
                A Communal Living Management App
              </h3>,
            ],
            "className": "logo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <h1>
                  Tribe
                </h1>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Tribe",
                },
                "ref": null,
                "rendered": "Tribe",
                "type": "h1",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "A Communal Living Management App",
              },
              "ref": null,
              "rendered": "A Communal Living Management App",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="user-stuff-nav"
              >
                <Link
                  className="home active"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
                <Link
                  className="manage-house "
                  replace={false}
                  to="/houselist"
                >
                  Manage House
                </Link>
                <Link
                  className="my-lists "
                  replace={false}
                  to="/userlist"
                >
                  My Lists
                </Link>
                <Link
                  className="leave-house"
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  Leave House
                </Link>
                <Link
                  className="signout"
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  Sign Out
                </Link>
              </div>,
              <p
                className="greeting"
              >
                Welcome to Dog9,  Amanda Brenner.
              </p>,
            ],
            "className": "user-stuff",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    className="home active"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>,
                  <Link
                    className="manage-house "
                    replace={false}
                    to="/houselist"
                  >
                    Manage House
                  </Link>,
                  <Link
                    className="my-lists "
                    replace={false}
                    to="/userlist"
                  >
                    My Lists
                  </Link>,
                  <Link
                    className="leave-house"
                    onClick={[Function]}
                    replace={false}
                    to="/"
                  >
                    Leave House
                  </Link>,
                  <Link
                    className="signout"
                    onClick={[Function]}
                    replace={false}
                    to="/"
                  >
                    Sign Out
                  </Link>,
                ],
                "className": "user-stuff-nav",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "className": "home active",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Manage House",
                    "className": "manage-house ",
                    "replace": false,
                    "to": "/houselist",
                  },
                  "ref": null,
                  "rendered": "Manage House",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "My Lists",
                    "className": "my-lists ",
                    "replace": false,
                    "to": "/userlist",
                  },
                  "ref": null,
                  "rendered": "My Lists",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Leave House",
                    "className": "leave-house",
                    "onClick": [Function],
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Leave House",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Out",
                    "className": "signout",
                    "onClick": [Function],
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Sign Out",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Dog9,  Amanda Brenner.",
                "className": "greeting",
              },
              "ref": null,
              "rendered": "Welcome to Dog9,  Amanda Brenner.",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
